### PR TITLE
Feature mtc 354 open position report

### PR DESCRIPTION
**Pull request by GSailen**

### CODE REVIEW

File: src/MarginTrading.AzureRepositories/AccountStatRepository.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            _dateService = dateService;
+        }
+        
+        public async Task Dump(IEnumerable<MarginTradingAccount> accounts)
+        {
+            var reportTime = _dateService.Now();
+            var entities = accounts.Select(x => AccountStatEntity.Create(x, reportTime));
+
+            var current = await _tableStorage.GetDataAsync();
+            await _tableStorage.DeleteAsync(current);
```
Comment (andrey-anokhin) [inline src/MarginTrading.AzureRepositories/AccountStatRepository.cs]: May be it is more efficient to remove and recreate table?

File: src/MarginTrading.AzureRepositories/AccountStatRepository.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+using MarginTrading.Backend.Core;
+using MarginTrading.Backend.Core.Repositories;
+using MarginTrading.Common.Services;
+
+namespace MarginTrading.AzureRepositories
+{
+    public class AccountStatRepository : IAccountStatRepository
+    {
+        private readonly INoSQLTableStorage<AccountStatEntity> _tableStorage;
+        private readonly ILog _log;
```
Comment (andrey-anokhin) [inline src/MarginTrading.AzureRepositories/AccountStatRepository.cs]: Not used

File: src/MarginTrading.AzureRepositories/OpenPositionsRepository.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+using Common.Log;
+using Lykke.SettingsReader;
+using MarginTrading.AzureRepositories.Entities;
+using MarginTrading.Backend.Core.Orders;
+using MarginTrading.Backend.Core.Repositories;
+using MarginTrading.Common.Services;
+
+namespace MarginTrading.AzureRepositories
+{
+    public class OpenPositionsRepository : IOpenPositionsRepository
```
Comment (andrey-anokhin) [inline src/MarginTrading.AzureRepositories/OpenPositionsRepository.cs]: Same comments, like for AccountStatRepository

File: src/MarginTrading.Backend.Core/MarginTrading.Backend.Core.csproj [hunk 4]
```diff
@@ -16,6 +16,7 @@
     <PackageReference Include="Lykke.SettingsReader" Version="2.4.0" />
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="..\MarginTrading.Backend.Contracts\MarginTrading.Backend.Contracts.csproj" />
```
Comment (andrey-anokhin) [inline src/MarginTrading.Backend.Core/MarginTrading.Backend.Core.csproj]: I think Core should not reference Contracts. Moreover it is not needed here )

File: src/MarginTrading.Backend.Core/Orders/IPosition.cs [hunk 5]
```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+
+namespace MarginTrading.Backend.Core.Orders
+{
+    public interface IPosition
```
Comment (andrey-anokhin) [inline src/MarginTrading.Backend.Core/Orders/IPosition.cs]: Why we need this interface and if really need, why there are commented fields here?

File: src/MarginTrading.Backend.Services/Services/ReportService.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            _openPositionsRepository = openPositionsRepository;
+            _accountStatRepository = accountStatRepository;
+        }
+        
+        public async Task DumpReportData()
+        {
+            var positions = _ordersCache.GetPositions();
+            await _openPositionsRepository.Dump(positions);
+            
+            var accountStat = _accountsCacheService.GetAll();
```
Comment (andrey-anokhin) [inline src/MarginTrading.Backend.Services/Services/ReportService.cs]: To have more consistent data it is better to get positions and account stats and only after this, dump data to the repo.

File: src/MarginTrading.SqlRepositories/Entities/OpenPositionEntity.cs [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        public decimal TotalPnL { get; set; }
+        public decimal ChargedPnL { get; set; }
+        public decimal Margin { get; set; }
+
+        List<RelatedOrderInfo> IPosition.RelatedOrders => RelatedOrders.DeserializeJson<List<RelatedOrderInfo>>();
+        public string RelatedOrders { get; set; }
+        
+        public DateTime HistoryTimestamp { get; set; }
+        
+        public static OpenPositionEntity Create(Position position, DateTime now)
```
Comment (andrey-anokhin) [inline src/MarginTrading.SqlRepositories/Entities/OpenPositionEntity.cs]: If we've added IPosition interface it is better to use it here. But I think that there is not need and we can remove it.

File: src/MarginTrading.Backend.Core/Orders/IPosition.cs (hunk 5 above)
Comment (GSailen) [reply src/MarginTrading.Backend.Core/Orders/IPosition.cs]: A child of experiments ) Removed.

**Pull request merged**